vendor/views/vendor.py

```diff
@@ -151,7 +143,7 @@ def post(self, request, *args, **kwargs):
 
             request.session['session_cart'] = session_cart
         else:
-            profile = self.request.user.customer_profile.get(site=get_current_site(self.request))      # Make sure they have a cart
+            profile = self.request.user.customer_profile.get(site=request.site)      # Make sure they have a cart
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(request)`